modules/jdbc/src/integrationTest/java/org/apache/ignite/jdbc/ItJdbcTransactionTest.java

```diff
@@ -0,0 +1,307 @@
+    /** Insert query. */
+    private static final String SQL_INSERT = "insert into TEST values (%d, '%s')";
+
+    /** Insert query for a prepared statement. */
+    private static final String SQL_INSERT_PREPARED = "insert into TEST values (?, ?)";
+
+    @BeforeAll
+    public static void beforeClass() throws Exception {
+        try (Statement statement = conn.createStatement()) {
+            statement.executeUpdate("create table TEST(ID int primary key, NAME varchar(20));");
+        }
+    }
+
+    @AfterEach
+    public void afterEach() throws Exception {
+        try (Statement statement = conn.createStatement()) {
+            statement.executeUpdate("delete from TEST;");
+        }
+
+        conn.setAutoCommit(true);
+    }
+
+    /**
+     * Tests {@link Connection#setAutoCommit(boolean)} and {@link Connection#getAutoCommit()} methods behaviour.
+     * <ul>
+     * <li>Auto-commit mode is enabled by default.
+     * <li>If {@code setAutoCommit} is called and the auto-commit mode is not changed, the call is a no-op.
+     * <li>If {@code setAutoCommit} is called during transaction and auto-commit mode changed then active transaction is committed.
+     * <li>Call to {@code setAutoCommit} or {@code getAutoCommit} on a closed connection throws exception.
+     * </ul>
+     *
+     * @throws SQLException If failed.
+     */
+    @Test
+    public void testAutoCommitModeChange() throws SQLException {
+        try (Connection conn = DriverManager.getConnection(URL)) {
+            assertTrue(conn.getAutoCommit());
+            conn.setAutoCommit(true);
+
+            conn.setAutoCommit(false);
+            conn.setAutoCommit(false);
+
+            int expRowsCount = 3;
+            String sqlUpdate = "insert into TEST (ID) values (1), (2), (3)";
+
+            try (Statement stmt = conn.createStatement()) {
+                conn.setAutoCommit(false);
+
+                assertEquals(expRowsCount, stmt.executeUpdate(sqlUpdate));
+                assertEquals(expRowsCount, rowsCount(conn));
+                assertEquals(0, rowsCount(AbstractJdbcSelfTest.conn));
+
+                // Verify that switching to auto-commit mode again commits started transaction.
+                conn.setAutoCommit(true);
+                assertEquals(expRowsCount, rowsCount(AbstractJdbcSelfTest.conn));
+            }
+
+            conn.close();
+
+            // Exception when called on closed connection
+            checkConnectionClosed(conn::getAutoCommit);
+            checkConnectionClosed(() -> conn.setAutoCommit(true));
+            checkConnectionClosed(() -> conn.setAutoCommit(false));
+        }
+    }
+
+    /**
+     * Tests {@link Connection#commit()} method behaviour.
+     *
+     * <p>Calling {@code commit} is expected to throw an exception if called in auto-commit mode or on a closed connection,
+     *
+     * @throws SQLException If failed.
+     */
+    @Test
+    public void testCommit() throws Exception {
```

Comment:
this test should be an unit test

Reply:
These tests (commit/rollback/autocommit) have been moved from ItJdbcConnectionSelfTest because you suggested moving all related to jdbc transaction tests to a separate test suite.
Personally, I don't see a big problem with these tests sitting in an integration suite (they don't take much time).
What are you suggesting to do with them - create a new Jdbc unit test with them  or leave them in ItJdbcConnectionSelfTest or move them to JdbcQueryEventHandlerImplTest?

Reply:
This test has been moved to `JdbcQueryEventHandlerImplTest`

Reply:
I wonder how you chose a new place for this test

Reply:
In order to add this test to ignite-jdbc, we need to mock the JdbcQueryEventHandler (connect() and finishTxAsync() methods) and therefore we will need to add the mockito dependency. And all of this just to check that the two methods are throwing an exception.
It seemed to me that this would already be a copy-paste, and among other things, in the JdbcQueryEventHandler test, I can check that extra transactions do not start.

And as I said before, I don't see any major issues with keeping these tests in the integration test suite, so I reverted this change for now.